pages/api/sessions.ts

```diff
@@ -26,16 +21,21 @@ export default async function handler(
 
   // get the data from the body of the request
   try {
-    const session: StripeSession = await stripeClient.checkout.sessions.create({
+    const session = await stripeClient.checkout.sessions.create({
       payment_method_types: ['card'],
       mode: request.body.mode,
       line_items: [
         { price: request.body.priceId, quantity: request.body.quantity },
       ],
-      success_url: successUrl + '?session_id={CHECKOUT_SESSION_ID}',
+      success_url: successUrl + '?sessionId={CHECKOUT_SESSION_ID}',
       cancel_url: cancelUrl,
     });
-    response.status(200).json({ session: session });
+
+    if (!session.url) {
+      return response.status(400).json({ error: 'Invalid Session URL' });
```

Comment:
not absolutely necessary, but one thing that I've been starting to do is write `return` on a different line than `response.status()...` so that it's clear to students that they don't need to write them together (it is performing two things - creating a response and then stopping execution):

```js
response.status(400).json({ error: 'Invalid Session URL' });
return;
```

Reply:
I think right now we must return the response right?

Reply:
with App Routes I guess so? since it's a constructor

```tsx
return NextResponse.json({ ... })
```